### PR TITLE
Implement glowing data

### PR DESCRIPTION
**Pull request by ZephireNZ**
API: SpongePowered/SpongeAPI#1122

This implements the new 1.9 Glowing API

Tests:

```
SOON
```

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeGlowingData.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.spongepowered.common.data.manipulator.immutable.entity;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeGlowingData.java]: Missing license headers.

File: src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+public class GlowingDataProcessor extends AbstractEntitySingleDataProcessor<Entity, Boolean, Value<Boolean>, GlowingData, ImmutableGlowingData> {
+
+    public GlowingDataProcessor() {
+        super(Entity.class, Keys.GLOWING);
+    }
+
+    @Override
+    protected boolean set(Entity dataHolder, Boolean value) {
+        dataHolder.setGlowing(checkNotNull(value));
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java]: No need to check the value isn't null.

File: src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        return true;
+    }
+
+    @Override
+    protected Optional<Boolean> getVal(Entity dataHolder) {
+        return Optional.of(dataHolder.isGlowing());
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java]: You can use `OptBool` here.

File: src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    protected ImmutableValue<Boolean> constructImmutableValue(Boolean value) {
+        return ImmutableSpongeValue.cachedOf(Keys.GLOWING, value, DataConstants.DEFAULT_GLOWING);
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java]: You can reference `this.key` (or `this.usedKey`) instead, if I remember correctly.

File: src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    protected Value<Boolean> constructValue(Boolean actualValue) {
+        return new SpongeValue<>(Keys.GLOWING, DataConstants.DEFAULT_GLOWING, actualValue);
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/processor/data/entity/GlowingDataProcessor.java]: You can reference `this.key` (or `this.usedKey`) instead, if I remember correctly.

**Pull request closed without merging**